dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx

```diff
@@ -254,9 +261,46 @@ export function PkgRepoForm(props: IPkgRepoFormProps) {
           );
         }
       }
+
+      // setting custom details for the Flux plugin
+      if (repo.packageRepoRef?.plugin?.name === PluginNames.PACKAGES_FLUX && repo.customDetail) {
+        const fluxPackageRepositoryCustomDetail =
+          repo.customDetail as Partial<FluxPackageRepositoryCustomDetail>;
+        setAuthProvider(fluxPackageRepositoryCustomDetail?.provider || "");
+      }
     }
   }, [repo, namespace, currentCluster, dispatch]);
 
+  const hasAuthProvider = useCallback(() => authProvider !== "", [authProvider]);
+
+  const handleFluxAuthProviderAuthChange = useCallback(() => {
+    setAuthMethod(PackageRepositoryAuth_PackageRepositoryAuthType.UNRECOGNIZED);
+    setShowAuthProviderDetails(true);
+    if (!hasAuthProvider()) {
+      setAuthProvider("generic");
+    }
+  }, [setAuthMethod, setShowAuthProviderDetails, hasAuthProvider, setAuthProvider]);
```

Comment:
It does make perfect sense to handle this into `handleAuthRadioButtonChange`, but the first line of it does `setAuthMethod(PackageRepositoryAuth_PackageRepositoryAuthType[e.target.value]);`, and the `PackageRepositoryAuth_PackageRepositoryAuthType` enum is from the generated source code and can't handle any other value AFAIK.

So either way it is going to be an ugly solution, because visually the radiobutton appears to be just another auth type. But it is not, therefore all the "special cases" like the mentioned `useEffect`. We are keeping two separate values that will end up in two different variables in the API DTO.
I don't like very much this solution, the alternative would be to change the API created by Greg to treat the `spec.provider` just as another auth type. WDYT?